annotations/static/annotations/js/annotators/text.js

```diff
@@ -327,24 +329,124 @@ DateAppellationCreator = {
+            counts: [],
+            conceptsFinal: [],
+            appell: this.appellations
+
+        }
+    },
+    template: `<div class="appellation-creator" style="max-height: 50vh; overflow-y: scroll;">
+                <concept-picker-item
+                    v-on:selectconcept="selectConcept"
+                    v-bind:concept=concept
+                    v-for="concept in conceptsFinal">
+                </concept-picker-item>
+               </div>`,
+    methods: {
+        selectConcept: function(concept) {
+            // Clear the concept search results.
+            this.concepts = [];
+            this.$emit('selectconcept', concept);
+        },
+        merge: function (appellations) {
+            this.conceptsFinal = [];
+            this.appell = appellations;
+            var appellationsLength =  this.appellations.length;
+            // remove duplicates and record count of each. Makes and new array with uri, index, count.
+            for (i = 0; i < appellationsLength; i++) {
+                if (this.concepts.includes(this.appellations[i].interpretation.uri)) {
+                    idx = this.concepts.indexOf(this.appellations[i].interpretation.uri);
+                    var toFind = idx + 2;
+                    this.concepts[toFind] = this.concepts[toFind] + 1;
+
+                } else{
+                    this.concepts.push(this.appellations[i].interpretation.uri, i, 1);
+                    }
+                }
+            var conceptsLength = this.concepts.length;
+            // adds objects back into an array using their indexes from the array made above
+            for (i = 1; i < conceptsLength; i = i + 3) {
+                this.appellationsCopy.push(this.appellations[this.concepts[i]]);
+            }
+            var appellationsCopyLength = this.appellationsCopy.length;
+            // add count to filtered objects
+            for (i = 0; i < appellationsCopyLength; i++) {
+                // find index of count in array.
+                if (i === 0) {
+                    idx = 2;
+                } else { 
+                    idx = (idx + 3);
+                }
+                this.appellationsCopy[i]["count"] = this.concepts[idx];
+                if (i <= 3) { // add only the first 4 concepts to the list
+                    this.conceptsFinal.push(this.appellationsCopy[i]);
+                }
+            }
+            // sort appellations copy by count
+            this.appellationsCopy.sort(function (a, b) {
+                return b.count - a.count;
+              });
+              // filter out duplicates if concept count list and recent concept list contain the same concepts
+              for(i = 0; i <= 3; i++) {
+                if (this.conceptsFinal.includes(this.appellationsCopy[i])){
+                } else {
+                    this.conceptsFinal.push(this.appellationsCopy[i]);
+                }
+              }
+            return this.conceptsFinal;
```

Comment:
maybe i'm missing something, but is the goal of this method to get a list of all the appellations sorted by count? if so, this looks a little too complicated. Why don't you create maps of the form {uri: count} and {uri: appellation} and then use the count map to create a list out of the appellations map?